### PR TITLE
feat(inputs.netflow): Allow to specify custom PEN field mappings

### LINKED ISSUE #13353: Handle PEN messages correctly
### Use Case

 Error processing IPFIX messages using telegraf [[inputs.netflow]] #13305. The question of interpreting the content of IPFIX messages with PEN fields remains open.

### Expected behavior

When receiving IPFIX messages containing PEN numbers are not interpreted as ntop fields with the corresponding content

### Actual behavior

When receiving IPFIX messages containing PEN numbers are not interpreted as 
ntop fields with the corresponding content[IPFIXCGS.pcap.txt](https://github.com/influxdata/telegraf/files/11579148/IPFIXCGS.pcap.txt)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #13353 

This PR adds the machinery to specify custom field mappings for "Private Enterprise Number" (aka vendor-specific) elements. To do so, you need to specify the mapping file in `private_enterprise_number_files`. The mappings are CSV files, an example can be found in the `mappings_ipfix` subdirectory.

### CODE REVIEW

**Comment by vknyazhev** [comment]
**Hi!downloaded amd64.deb, started the installation.Is it normal that it is installed like this?the last patch was installed without errors**
@home:~$ sudo dpkg -i telegraf_1.27.0~476b7765-0_amd64.deb 
[sudo] password for vknyazhev: 
Sorry, try again.
[sudo] password for vknyazhev: 
(Reading database ... 111536 files and directories currently installed.)
Preparing to unpack telegraf_1.27.0~476b7765-0_amd64.deb ...
Unpacking telegraf (1.27.0-0) over (1.27.0-0) ...
Setting up telegraf (1.27.0-0) ...
touch: setting times of '/var/lib/telegraf/statefile': No such file or directory
chown: cannot access '/var/lib/telegraf/statefile': No such file or directory
chmod: cannot access '/var/lib/telegraf/statefile': No such file or directory
@home:~$ 
**And the log shows this:**
@home:~$ tail -f /var/log/telegraf/telegraf.log
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 495 35632 [3 3]}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 180 35632 []}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 360 35632 []}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 181 35632 [0 0]}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 182 35632 []}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 184 35632 []}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 187 35632 [109 97 105 108 46 114 117]}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 361 35632 [109 97 105 108 46 114 117]}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 460 35632 []}
2023-05-31T09:02:40Z D! [inputs.netflow] unknown PEN data field {true 461 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] received 228 bytes
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 123 35632 [0 0 0 2]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 124 35632 [0 0 0 11]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 125 35632 [0 0 0 23]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 127 35632 [0 0 0 0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 109 35632 [0 0 0 0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 128 35632 [0 0 0 0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 110 35632 [0 0 0 0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 118 35632 [0 63]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 188 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 489 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 490 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 493 35632 [0 0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 494 35632 [0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 495 35632 [0 0]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 180 35632 [114 51 46 111 46 108 101 110 99 114 46 111 114 103 47]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 360 35632 [80 79 83 84]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 181 35632 [0 200]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 182 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 184 35632 [97 112 112 108 105 99 97 116 105 111 110 47 111 99 115 112 45 114 101 115 112 111 110 115 101]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 187 35632 [114 51 46 111 46 108 101 110 99 114 46 111 114 103]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 361 35632 [108 101 110 99 114 46 111 114 103]}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 460 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] unknown PEN data field {true 461 35632 []}
2023-05-31T09:02:41Z D! [inputs.netflow] received 152 bytes

**Comment by srebhan** [comment]
Pushed another version with more debugging information. Can you please test and show me the complete log?

Furthermore, you don't necessarily need to **install** the PR binary, you can also just use the [linux_amd64.tar.gz](https://output.circle-artifacts.com/output/job/036aa3a9-3ad1-49b2-9057-1bcd7bf73bf1/artifacts/0/build/dist/telegraf-1.27.0~476b7765_linux_amd64.tar.gz) archive, extract it and run the single static binary in there with your config manually e.g. via `./telegraf --config <your config> --debug`. Strip down your config to only `inputs.netflow` and debug output e.g.
```toml
[[inputs.netflow]]
   ...

[[outputs.file]]
  files = ["stdout"]
```

**Comment by srebhan** [comment]
@vknyazhev your file is called `nop-35632.csv` (missing `t`) but Telegraf reads `ntop-35632.csv`, so it might be right that the file is missing... ;-)

File: plugins/inputs/netflow/netflow_decoder.go [hunk 1]
```diff
@@ -693,6 +701,20 @@ func (d *netflowDecoder) decodeValueIPFIX(field netflow.DataField) []telegraf.Fi
 		elementID = field.Type & (0x4000 ^ 0xffff)
 	}
 
+	// Handle messages with Private Enterprise Numbers (PENs)
+	if field.PenProvided {
+		key := fmt.Sprintf("%d.%d", field.Pen, elementID)
+		// REMOVE
+		d.Log.Debugf("decoding PEN field %+v (eid: %d; key: %q)", field, elementID, key)
```
Comment (powersj) [inline plugins/inputs/netflow/netflow_decoder.go:708]: Is this in here to aid debugging and were you going to remove it once you had confirmation from the user?

File: plugins/inputs/netflow/netflow_decoder.go [hunk 2]
```diff
@@ -693,6 +701,20 @@ func (d *netflowDecoder) decodeValueIPFIX(field netflow.DataField) []telegraf.Fi
...
+		if m, found := d.mappingsPEN[key]; found {
+			name := prefix + m.name
+			return []telegraf.Field{{Key: name, Value: m.decoder(raw)}}
+		}
+		d.Log.Debugf("unknown IPFIX PEN data field %v", field)
```
Comment (powersj) [inline plugins/inputs/netflow/netflow_decoder.go:713]: Same question with this debug

File: plugins/inputs/netflow/netflow_decoder.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/netflow/netflow_decoder.go:708]: This one is for removal as it is debugging only...

**Comment by srebhan** [comment]
@powersj would love to get your approval and hold-off merging this for user-feedback...

**Comment by srebhan** [comment]
@vknyazhev but now you did not load the PEN mappings... You need to add `private_enterprise_number_files = ["/path/to/ntop-35632.csv"]` to your config...

**Comment by vknyazhev** [comment]
@srebhan the fact of the matter is that the file has been added
FilehasbeenAdded

**Comment by srebhan** [comment]
Are you using the latest binary in the PR? It should print out how many mappings it loaded when running with `--debug`... Can you check if you see some info in the logfile or something?

**Comment by vknyazhev** [comment]
@srebhan Hello, yes I am using the latest PR. I downloaded your archive, unpacked it, and launched the telegraph from there. The log was attached
[dnsperf-2(10.170.1.202)--10-05.log](https://github.com/influxdata/telegraf/files/11673659/dnsperf-2.10.170.1.202.--10-05.log)

**Comment by srebhan** [comment]
@vknyazhev you are not using the latest version in this PR. Your log says
> 10:08:36.533 $2023-06-07T07:08:35Z I! Starting Telegraf 1.27.0-476b7765

while the binary is [1.27.0~4ef86ddd](https://output.circle-artifacts.com/output/job/913b6589-78cc-418d-a0ad-9b96790d4bce/artifacts/0/build/dist/telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz). It's probably because you start the Telegraf in your PATH and not the one in the current directory...

**Comment by vknyazhev** [comment]
**Downloaded:**
@home:~$ wget https://output.circle-artifacts.com/output/job/913b6589-78cc-418d-a0ad-9b96790d4bce/artifacts/0/build/dist/telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz
--2023-06-07 09:31:30--  https://output.circle-artifacts.com/output/job/913b6589-78cc-418d-a0ad-9b96790d4bce/artifacts/0/build/dist/telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz
Resolving output.circle-artifacts.com (output.circle-artifacts.com)... 13.32.99.80, 13.32.99.7, 13.32.99.41, ...
... 200 OK
Length: 51900148 (49M) [application/gzip]
Saving to: вЂtelegraf-1.27.0~4ef86ddd_linux_amd64.tar.gzвЂ™

telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz        17%[=================>                                                                                          ]   8.58M   283KB/s    eta 2m 45s
**Unpacked**:
@home:~$ tar -xvf telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz 
./
./telegraf-1.27.0/
./telegraf-1.27.0/usr/
./telegraf-1.27.0/usr/bin/
./telegraf-1.27.0/usr/bin/telegraf
./telegraf-1.27.0/usr/lib/
./telegraf-1.27.0/usr/lib/telegraf/
./telegraf-1.27.0/usr/lib/telegraf/scripts/
./telegraf-1.27.0/usr/lib/telegraf/scripts/telegraf.service
./telegraf-1.27.0/usr/lib/telegraf/scripts/init.sh
./telegraf-1.27.0/etc/
./telegraf-1.27.0/etc/logrotate.d/
./telegraf-1.27.0/etc/logrotate.d/telegraf
./telegraf-1.27.0/etc/telegraf/
./telegraf-1.27.0/etc/telegraf/telegraf.d/
./telegraf-1.27.0/etc/telegraf/telegraf.conf
./telegraf-1.27.0/var/
./telegraf-1.27.0/var/log/
./telegraf-1.27.0/var/log/telegraf/
vknyazhev@home:~$
**Launching**
vknyazhev@home:~$ cd telegraf-1.27.0/etc/telegraf/telegraf.d/
vknyazhev@home:~/telegraf-1.27.0/etc/telegraf/telegraf.d$ ls
vknyazhev@home:~/telegraf-1.27.0/etc/telegraf/telegraf.d$ sudo cp /home/vknyazhev/ntop-35632.csv ntop-35632.csv
vknyazhev@home:~/telegraf-1.27.0/etc/telegraf/telegraf.d$ ls
ntop-35632.csv
vknyazhev@home:~/telegraf-1.27.0/etc/telegraf/telegraf.d$ sudo telegraf --config /home/vknyazhev/ipfix_test.conf --debug
2023-06-07T09:35:59Z I! Loading config: /home/vknyazhev/ipfix_test.conf
2023-06-07T09:35:59Z I! Starting Telegraf 1.27.0-476b7765
2023-06-07T09:35:59Z I! Available plugins: 236 inputs, 9 aggregators, 28 processors, 23 parsers, 58 outputs, 4 secret-stores
2023-06-07T09:35:59Z I! Loaded inputs: netflow
2023-06-07T09:35:59Z I! Loaded aggregators: 
2023-06-07T09:35:59Z I! Loaded processors: 
2023-06-07T09:35:59Z I! Loaded secretstores: 
2023-06-07T09:35:59Z I! Loaded outputs: file
2023-06-07T09:35:59Z I! Tags enabled: host=home
2023-06-07T09:35:59Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"home", Flush Interval:10s
2023-06-07T09:35:59Z D! [agent] Initializing plugins
2023-06-07T09:35:59Z D! [agent] Connecting outputs
2023-06-07T09:35:59Z D! [agent] Attempting connection to [outputs.file]
2023-06-07T09:35:59Z D! [agent] Successfully connected to outputs.file
2023-06-07T09:35:59Z D! [agent] Starting service inputs
2023-06-07T09:35:59Z I! [inputs.netflow] Listening on udp://[::]:2055
2023-06-07T09:36:09Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2023-06-07T09:36:19Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2023-06-07T09:36:29Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2023-06-07T09:36:31Z D! [inputs.netflow] received 288 bytes
2023-06-07T09:36:31Z W! [inputs.netflow] No info template 261 found for and domain id 16; skipping packet
2023-06-07T09:36:32Z D! [inputs.netflow] received 272 bytes
2023-06-07T09:36:32Z W! [inputs.netflow] No info template 261 found for and domain id 16; skipping packet
2023-06-07T09:36:32Z D! [inputs.netflow] received 268 bytes
2023-06-07T09:36:32Z W! [inputs.netflow] No info template 261 found for and domain id 16; skipping packet
2023-06-07T09:36:32Z D! [inputs.netflow] received 980 bytes
2023-06-07T09:36:32Z D! [inputs.netflow] received 588 bytes
2023-06-07T09:36:32Z D! [inputs.netflow] received 268 bytes
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 123 35632 [0 0 0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 124 35632 [0 0 0 2]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 125 35632 [0 0 0 9]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 127 35632 [0 0 0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 109 35632 [0 0 0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 128 35632 [0 0 0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 110 35632 [0 0 0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 118 35632 [0 7]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 188 35632 []}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 489 35632 []}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 490 35632 []}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 493 35632 [0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 494 35632 [0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 495 35632 [0 0]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 180 35632 [119 119 119 46 104 105 109 107 105 45 101 100 117 46 114 117 47 109 99 101 45 100 97 116 97 47 114 101 115 104 46 103 105 102]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 360 35632 [71 69 84]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 181 35632 [0 200]}
2023-06-07T09:36:32Z D! [inputs.netflow] unknown PEN data field {true 182 35632 [119 119 119 46 104 105 109 107 105 45 101 100 117 46 114 117 47 103 101 110 101 114 97 108 47 105 110 100 101 120 46 100 111]}
**Or so**
vknyazhev@home:~/telegraf-1.27.0/etc/telegraf/telegraf.d$  sudo cp /home/vknyazhev/ipfix_test.conf ipfix_test.conf
vknyazhev@home:~/telegraf-1.27.0/etc/telegraf/telegraf.d$ sudo telegraf --config ipfix_test.conf --debug
2023-06-07T09:39:33Z I! Loading config: ipfix_test.conf
2023-06-07T09:39:33Z I! Starting Telegraf 1.27.0-476b7765
2023-06-07T09:39:33Z I! Available plugins: 236 inputs, 9 aggregators, 28 processors, 23 parsers, 58 outputs, 4 secret-stores
2023-06-07T09:39:33Z I! Loaded inputs: netflow
2023-06-07T09:39:33Z I! Loaded aggregators: 
2023-06-07T09:39:33Z I! Loaded processors: 
2023-06-07T09:39:33Z I! Loaded secretstores: 
2023-06-07T09:39:33Z I! Loaded outputs: file
2023-06-07T09:39:33Z I! Tags enabled: host=home
2023-06-07T09:39:33Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"home", Flush Interval:10s
2023-06-07T09:39:33Z D! [agent] Initializing plugins
2023-06-07T09:39:33Z D! [agent] Connecting outputs
2023-06-07T09:39:33Z D! [agent] Attempting connection to [outputs.file]
2023-06-07T09:39:33Z D! [agent] Successfully connected to outputs.file
2023-06-07T09:39:33Z D! [agent] Starting service inputs
2023-06-07T09:39:33Z I! [inputs.netflow] Listening on udp://[::]:2055
2023-06-07T09:39:41Z D! [inputs.netflow] received 184 bytes
2023-06-07T09:39:41Z W! [inputs.netflow] No info template 261 found for and domain id 16; skipping packet
2023-06-07T09:39:41Z D! [inputs.netflow] received 160 bytes
2023-06-07T09:39:41Z W! [inputs.netflow] No info template 261 found for and domain id 16; skipping packet
2023-06-07T09:39:41Z D! [inputs.netflow] received 188 bytes

**Comment by vknyazhev** [comment]
Maybe the wrong version is in the archive?

**Comment by srebhan** [comment]
@vknyazhev you are running ` sudo telegraf --config ipfix_test.conf --debug` which used the Telegraf **installed** in your path. Try `which telegraf` to see which one is used.
To use the downloaded version run
```shell
# sudo ~/telegraf-1.27.0/usr/bin/telegraf--config ipfix_test.conf --debug
```
you actually should run

**Comment by vknyazhev** [comment]
**Still something with the version in the archive**
vknyazhev@home:~$ tar -zxvf telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz 
./
./telegraf-1.27.0/
./telegraf-1.27.0/usr/
./telegraf-1.27.0/usr/bin/
./telegraf-1.27.0/usr/bin/telegraf
./telegraf-1.27.0/usr/lib/
./telegraf-1.27.0/usr/lib/telegraf/
./telegraf-1.27.0/usr/lib/telegraf/scripts/
./telegraf-1.27.0/usr/lib/telegraf/scripts/telegraf.service
./telegraf-1.27.0/usr/lib/telegraf/scripts/init.sh
./telegraf-1.27.0/etc/
./telegraf-1.27.0/etc/logrotate.d/
./telegraf-1.27.0/etc/logrotate.d/telegraf
./telegraf-1.27.0/etc/telegraf/
./telegraf-1.27.0/etc/telegraf/telegraf.d/
./telegraf-1.27.0/etc/telegraf/telegraf.conf
./telegraf-1.27.0/var/
./telegraf-1.27.0/var/log/
./telegraf-1.27.0/var/log/telegraf/
vknyazhev@home:~$ ls
influxdata-archive_compat.key  influxdata-archive_compat.key.1  ipfix_test.conf  ntop-35632.csv  pen.txt  SW  telegraf-1.27.0  telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz  token.txt  wget-log
vknyazhev@home:~$ cd telegraf-1.27.0/
etc/ usr/ var/ 
vknyazhev@home:~$ cd telegraf-1.27.0/usr/
bin/ lib/ 
vknyazhev@home:~$ cd telegraf-1.27.0/usr/bin/
vknyazhev@home:~/telegraf-1.27.0/usr/bin$ ls
telegraf
**vknyazhev@home:~/telegraf-1.27.0/usr/bin$ sudo telegraf --version
[sudo] password for vknyazhev: 
**Telegraf 1.27.0-476b7765** (git: pull/13360@476b7765)**
vknyazhev@home:~/telegraf-1.27.0/usr/bin$ sudo cp /home/vknyazhev/ipfix_test.conf ipfix_test.conf
vknyazhev@home:~/telegraf-1.27.0/usr/bin$ sudo cp /home/vknyazhev/ntop-35632.csv ntop-35632.csv
vknyazhev@home:~/telegraf-1.27.0/usr/bin$ sudo te
tee            telegraf       telinit        telnet         telnet.netkit  tempfile       test           
vknyazhev@home:~/telegraf-1.27.0/usr/bin$ sudo telegraf --config ipfix_test.conf --debug
2023-06-07T10:34:26Z I! Loading config: ipfix_test.conf
2023-06-07T10:34:26Z I! Starting Telegraf 1.27.0-476b7765
2023-06-07T10:34:26Z I! Available plugins: 236 inputs, 9 aggregators, 28 processors, 23 parsers, 58 outputs, 4 secret-stores
2023-06-07T10:34:26Z I! Loaded inputs: netflow
2023-06-07T10:34:26Z I! Loaded aggregators: 
2023-06-07T10:34:26Z I! Loaded processors: 
2023-06-07T10:34:26Z I! Loaded secretstores: 
2023-06-07T10:34:26Z I! Loaded outputs: file
2023-06-07T10:34:26Z I! Tags enabled: host=home
2023-06-07T10:34:26Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"home", Flush Interval:10s
2023-06-07T10:34:26Z D! [agent] Initializing plugins
2023-06-07T10:34:26Z D! [agent] Connecting outputs
2023-06-07T10:34:26Z D! [agent] Attempting connection to [outputs.file]
2023-06-07T10:34:26Z D! [agent] Successfully connected to outputs.file
2023-06-07T10:34:26Z D! [agent] Starting service inputs
2023-06-07T10:34:26Z I! [inputs.netflow] Listening on udp://[::]:2055
2023-06-07T10:34:26Z D! [inputs.netflow] received 184 bytes
2023-06-07T10:34:26Z W! [inputs.netflow] No info template 261 found for and domain id 16; skipping packet
2023-06-07T10:34:26Z D! [inputs.netflow] received 980 bytes
2023-06-07T10:34:26Z D! [inputs.netflow] received 588 bytes
2023-06-07T10:34:26Z D! [inputs.netflow] received 280 bytes
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 123 35632 [0 0 0 0]}
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 124 35632 [0 0 0 3]}
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 125 35632 [0 0 0 7]}
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 127 35632 [0 0 0 0]}
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 109 35632 [0 0 0 0]}
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 128 35632 [0 0 0 0]}
2023-06-07T10:34:26Z D! [inputs.netflow] unknown PEN data field {true 110 35632 [0 0 0 0]}

**Comment by powersj** [comment]
> Still something with the version in the archive

Unfortunately no. You are still running `telegraf` which will run the version installed on your system. You must preface your command with the `./` to run the local telegraf.

Here is exactly what you need to run:

```bash
tar -zxvf telegraf-1.27.0~4ef86ddd_linux_amd64.tar.gz 
cd telegraf-1.27.0/usr/bin/
sudo cp /home/vknyazhev/ipfix_test.conf ipfix_test.conf
sudo cp /home/vknyazhev/ntop-35632.csv ntop-35632.csv
./telegraf version
./telegraf --config ipfix_test.conf --debug
```

Those leading `./` tell the shell to run the local version of telegraf. Otherwise, your shell will look in your `$PATH`, which will default to binaries found in `/usr/bin` among other directories.

**Comment by powersj** [comment]
Thanks!

@srebhan how does that look?

**Comment by vknyazhev** [comment]
[debug.txt](https://github.com/influxdata/telegraf/files/11678221/debug.txt)

**Comment by srebhan** [comment]
@vknyazhev you are using the wrong mapping file. Please download the one from [ **`mappings_ipfix`**](https://raw.githubusercontent.com/influxdata/telegraf/4ef86ddd0dacf35a996fa4efa64cb59aeed58a7d/plugins/inputs/netflow/mappings_ipfix/ntop-35632.csv) and **NOT** the one from `mappings_netflow`! I added a check to detect those wrong files early...

You can check yourself by looking at the CSV file. The data lines should start with `35632.` followed by an element-ID...

**Pull request merged**